command/build.go

```diff
@@ -97,22 +98,46 @@ func (m *Meta) GetConfig(cla *MetaArgs) (packer.Handler, int) {
 		m.Ui.Error(fmt.Sprintf("%q: %s", cla.Path, err))
 		return nil, 1
 	}
+	cla.ConfigType = cfgType
 
 	switch cfgType {
 	case ConfigTypeHCL2:
 		// TODO(azr): allow to pass a slice of files here.
 		return m.GetConfigFromHCL(cla)
 	default:
-		// TODO: uncomment once we've polished HCL a bit more.
-		// c.Ui.Say(`Legacy JSON Configuration Will Be Used.
-		// The template will be parsed in the legacy configuration style. This style
-		// will continue to work but users are encouraged to move to the new style.
-		// See: https://packer.io/guides/hcl
-		// `)
 		return m.GetConfigFromJSON(cla)
 	}
 }
 
+//DisplayLegacyConfigWarning displays Packer defined legacy warnings
+// This is an internal-only method that should be called by commands that don't have
+// a different mode when handling legacy JSON templates; mainly build, validate.
+func DisplayLegacyConfigWarning(ui packersdk.Ui) {
+	if v := os.Getenv(PackerLegacyModeEnv); strings.ToUpper(v) == packerLegacyModeOn {
+		return
+	}
+
+	coloredUi := packer.ColoredUi{
+		Color: packer.UiColorRed,
+		Ui:    ui,
+	}
+
+	coloredUi.Say(`
+Legacy JSON Configuration Mode enabled.
+The template will be parsed in the legacy configuration style. Legacy style
+templates will continue to work but users are encouraged to move HCL style templates.
```

Comment:
Do we plan to remove JSON support entirely? If we do, maybe we can say that will continue to work for a limited time